src/psij/job.py

```diff
@@ -83,6 +83,10 @@ def native_id(self) -> Optional[str]:
         else:
             return str(self._native_id)
 
+    @native_id.setter
```

Comment:
We probably should document that this setter is for internal use,  *must* not be used by the application,  and that `attach` is to be used instead.

Reply:
That may not be necessary since it's reasonable to access protected properties from other classes in the package. That said, I've been conflating protected and private entities a bit, so maybe that should be fixed separately.

Reply:
I removed the setter.

Reply:
Yeah, Python is missing a `friend` mechanism...